Celeste.Mod.mm/Mod/Everest/Everest.Updater.cs

```diff
@@ -368,13 +372,24 @@ public class Source {
                     if (Type.GetType("Mono.Runtime") != null) {
                         installer.StartInfo.FileName = "mono";
                         installer.StartInfo.Arguments = $"\"{installerPath}\"";
-                        if (File.Exists("/bin/sh")) {
-                            installer.StartInfo.FileName = "/bin/sh";
-                            installer.StartInfo.Arguments = $"-c \"cd '{extractedPath}'; mono MiniInstaller.exe\"";
+                        if (File.Exists("/bin/bash")) {
+                            installer.StartInfo.FileName = "/bin/bash";
+                            installer.StartInfo.Arguments = $"-c \"cd '{extractedPath}'; unset MONO_PATH LD_LIBRARY_PATH LC_ALL MONO_CONFIG; /usr/bin/mono MiniInstaller.exe &> log.txt\"";
                         }
                     }
                     installer.StartInfo.WorkingDirectory = extractedPath;
-                    installer.Start();
+                    if (Environment.OSVersion.Platform == PlatformID.Unix) {
+                        installer.StartInfo.UseShellExecute = false;
+                        installer.StartInfo.RedirectStandardOutput = true;
+                        installer.OutputDataReceived += (sender, args) => progress.LogLine(args.Data);
+                        installer.Start();
+                        progress.LogLine("Patching the game in-place...");
```

Comment:
Patching the game in-place reportedly caused the game process to hang for some people, while for others, the installer was unable to write to the running assembly at all (just like on Windows). Is that still a concern to have with your in-place patching code?

Reply:
It shouldn't, but if that ends up being a problem again we could add a slight delay to the installer.